doc/develop/languages/c/picolibc.rst

```diff
@@ -0,0 +1,157 @@
+.. _c_library_picolibc:
+
+Picolibc
+########
+
+`Picolibc`_ is a complete C library implementation written for the
+embedded systems, targeting `C17 (ISO/IEC 9899:2018)`_ and `POSIX 2018
+(IEEE Std 1003.1-2017)`_ standards. Picolibc is an external open
+source project which is provided for Zephyr as a module, and included
+as part of the :ref:`toolchain_zephyr_sdk` in precompiled form for
+each supported architecture (:file:`libc.a`)
+
+.. note::
+   Picolibc is also available for other 3rd-party toolchains, such as
+   :ref:`toolchain_gnuarmemb`.
+
+Zephyr implements the “API hook” functions that are invoked by the C
+standard library functions in the Picolibc. These hook functions are
+implemented in :file:`lib/libc/picolibc/libc-hooks.c` and translate
+the library internal system calls to the equivalent Zephyr API calls.
+
+.. _`Picolibc`: https://github.com/picolibc/picolibc
+.. _`C17 (ISO/IEC 9899:2018)`: https://www.iso.org/standard/74528.html
+.. _`POSIX 2018 (IEEE Std 1003.1-2017)`: https://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.html
+
+Picolibc Module
+===============
+
+When built as a Zephyr module, there are several configuration knobs
+available to adjust the feature set in the library, balancing what the
+library supports versus the code size of the resulting
+functions. Because the standard C++ library must be compiled for the
+target C library, the module version cannot be used with applications
+which need that library. Building the module version will increase the
```

Comment:
I'm having trouble understanding the reference in `that library` -- do you mean the C++ or the C library? I think C++, right?

Reply:
Yeah, sorry for the ambiguous reference. I've made it explicit.